sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -208,23 +208,30 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
           }
           reader
 
-        case _ =>
-          throw new AnalysisException(s"$cls does not support data reading.")
+        case _ => null // fall back to v1
       }
 
-      Dataset.ofRows(sparkSession, DataSourceV2Relation(reader))
+      if (reader == null) {
```

Comment:
This section got more complicated with this. Add more comments above this section on this fallback policy.